Automattic/wp-calypso
# Move I am a developer toggle to Developer Page under `/me`

Related to https://github.com/Automattic/dotcom-forge/issues/5231 and https://github.com/Automattic/wp-calypso/pull/86648

## Proposed Changes

**NOTE: We are rolling out the Developer Features Page ("I am a developer" toggle + showcase a list of developer features) in one batch. The screenshots below are no longer relevant. See what the final state of the page looks like: https://github.com/Automattic/wp-calypso/pull/86648**


* Added "Developer Features" item to the sidebar.
* Moved "I am developer" toggle from "Profile" to "Developer Features".

| My Profile - Before | My Profile - After |
| ------ | ------ |
|<img width="1313" alt="Screen Shot 2024-01-17 at 8 25 16 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/dcd22998-7739-4790-8b98-2d304d405ac3">|<img width="1311" alt="Screen Shot 2024-01-17 at 8 26 25 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/6375a59a-a446-4ebb-8af8-0a2591d406e1">|

* Added a "Developer Features" page with the "I am a developer" toggle.

<img width="1207" alt="Screen Shot 2024-01-19 at 4 26 09 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/4e0c57b3-b4b8-4039-b9fe-9eac4efca98f">

* Track event `calypso_page_view` when user lands on the "Developer Features" page, with path `me/developer`.

<img width="588" alt="Screen Shot 2024-01-17 at 8 39 09 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/8c12e829-a0ff-45df-9ea3-363d823fad49">

* Track event `calypso_me_is_dev_account_toggled` when user updates the toggle. Send `enable: 1` when the toggle is on, `enable: 0` when the toggle is off.

<img width="1345" alt="Screen Shot 2024-01-17 at 8 31 32 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/ce0f5fb5-f77c-4c51-886d-f422017bb80b">

* Note that, I also took the opportunity to address a but where the save button on the "Profile" page isn't disabled during user settings update.

## Testing Instructions



* Run the direct build and navigate to `/me`
* Assert that the "I am a developer" toggle is removed from "Profile" and that profile details can be updated.
* Click "Developer Features" and assert that you see the Developer Features Page with the "I am a developer" toggle.
* Assert that you page tracking and click tracking work as described above.
* Assert that the "I am a developer" setting is updated properly.
